src/ansi-c/library/cprover_contracts.c

```diff
@@ -1418,4 +1422,36 @@ __CPROVER_HIDE:;
 #endif
   }
 }
-#endif
+
+/// \brief Implementation of the `obeys_contract` front-end predicate.
+/// \return True iff a function pointer points to the specified contract.
+///
+/// \details If called in an assumption context, translates to an assignment
+/// `function_pointer = contract`. If called in an assertion context, translates
+/// to an `function_pointer == contract`.
+__CPROVER_bool __CPROVER_contracts_obeys_contract(
+  void (**function_pointer)(void),
+  void (*contract)(void),
+  __CPROVER_contracts_write_set_ptr_t set)
+{
+__CPROVER_HIDE:;
+  __CPROVER_assert(
+    (set != 0) &
+      ((set->assume_requires_ctx == 1) | (set->assert_requires_ctx == 1) |
+       (set->assume_ensures_ctx == 1) | (set->assert_ensures_ctx == 1)),
+    "__CPROVER_obeys_contract is used only in requires or ensures clauses");
+  if((set->assume_requires_ctx == 1) | (set->assume_ensures_ctx == 1))
+  {
+    if(*function_pointer != 0)
```

Comment:
Did you mean `function_pointer != 0` instead of doing a dereference?

Reply:
function_pointer is a pointer to a pointer